github-runner-manager/src/github_runner_manager/jobmanager_api.py

```diff
@@ -0,0 +1,170 @@
+#  Copyright 2025 Canonical Ltd.
+#  See LICENSE file for licensing details.
+
+"""Module containing logic to handle calls to the jobmanager api."""
+from enum import Enum
+
+import jobmanager_client
+from jobmanager_client.exceptions import ApiException, NotFoundException
+from pydantic import BaseModel
+from urllib3.exceptions import RequestError
+
+
+class JobManagerAPIError(Exception):
+    """Base exception for JobManager API errors."""
+
+
+class JobManagerAPINotFoundError(JobManagerAPIError):
+    """Exception raised when a runner is not found in the JobManager API."""
+
+
+class JobStatus(str, Enum):
+    """Status of a job on the JobManager.
+
+    Attributes:
+        IN_PROGRESS: Represents a job that is in progress.
+        PENDING: Represents a job that is pending.
+    """
+
+    IN_PROGRESS = "IN_PROGRESS"
+    PENDING = "PENDING"
+
+
+class Job(BaseModel):
+    """Represents a job on the JobManagerAPI.
+
+    Attributes:
+        status: The status of the job.
+    """
+
+    status: str | None
+
+
+class RunnerStatus(str, Enum):
+    """Status of a runner on the JobManager.
+
+    Attributes:
+        IN_PROGRESS: Represents a job that is in progress.
+        PENDING: Represents a job that is pending.
+    """
+
+    IN_PROGRESS = "IN_PROGRESS"
+    PENDING = "PENDING"
+
+
+class RunnerRegistration(BaseModel):
+    """Represents a runner registration response from the JobManagerAPI.
+
+    Attributes:
+        id: The ID of the registered runner.
+        token: The token for the registered runner.
+    """
+
+    id: int
+    token: str
+
+
+class RunnerHealth(BaseModel):
+    """Represents the health status of a runner on the JobManagerAPI.
+
+    Attributes:
+        status: The health status of the runner.
+        deletable: Indicates if the runner can be deleted.
+    """
+
+    status: str
```

Comment:
idea would be to use the `RunnerStatus` enum, but right now, the openapi spec doesn't provide an enumeration, so I cannot foresee the potential values. The enum is still used in the jobmanager_provider to distinguish on the states.